pinot-controller/src/main/java/org/apache/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -156,12 +157,15 @@ protected void postprocess(Context context) {
    * Updates metrics related to the table config.
    * If table config not found, resets the metrics
    */
-  private void updateTableConfigMetrics(String tableNameWithType, TableConfig tableConfig) {
+  private void updateTableConfigMetrics(String tableNameWithType, TableConfig tableConfig, Context context) {
     if (tableConfig == null) {
       LOGGER.warn("Found null table config for table: {}. Resetting table config metrics.", tableNameWithType);
       _controllerMetrics.setValueOfTableGauge(tableNameWithType, ControllerGauge.REPLICATION_FROM_CONFIG, 0);
       return;
     }
+    if (tableConfig.isUpsertEnabled()) {
```

Comment:
Shall we also move `_offlineTableCount` and `_realTimeTableCount` metric update here?

Reply:
Done